src/lib/dom-api.html

```diff
@@ -593,6 +593,61 @@
 
     };
 
+    Object.defineProperties(DomApi.prototype, {
+      activeElement: {
+        configurable: true,
+        get: function() {
+          var ownerDocument = this.node.ownerDocument;
+
+          // Shadow DOM
+          if (Polymer.Settings.useShadow) {
+            if (ownerDocument || !window.ShadowDOMPolyfill) {
+              return this.node.activeElement;
+            } else {
+              return window.ShadowDOMPolyfill.wrapIfNeeded(this.node).activeElement;
+            }
+          }
+
+          // Shady DOM
+
+          var trueDocument = ownerDocument || this.node;
+
+          // Not a document or shady root.
+          if (ownerDocument && !this.node._isShadyRoot) {
+            return this.node.activeElement;
+          }
+
+          var active = trueDocument.activeElement;
+          if (!active) return null;
+
+          // `this.node` is either a document or shady root; iterate upwards
+          // to find the active element's most shallow host (relative to this
+          // node or its containing document).
+          var activeRoot = Polymer.dom(active).getOwnerRoot();
+          while (activeRoot && activeRoot !== this.node) {
+            active = activeRoot.host;
+            activeRoot = Polymer.dom(active).getOwnerRoot();
+          }
+
+          // This node is a document and we’ve reached it; we should return the
+          // active element's most shallow host.
+          if (!ownerDocument && !activeRoot) {
```

Comment:
(from offline discussion) `getOwnerRoot` returns null if the element is within the light DOM of the document, similar to how `ownerDocument` is null for documents. Also, I'll remove the use of `ownerDocument` and just compare with `document` directly for the sake of simplicity.